Assignment 2.Rmd

```diff
@@ -95,6 +107,41 @@ Now practice with data from our class. Please create a **person-network** with t
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+``` {r}
+library(tidyr)
+library(dplyr)
+D3 <- read.csv("hudk4050-classes.csv", header = TRUE, na.strings = "")
+
+# remove space from the name of the classes
+D4 <- data.frame(apply(D3, 2, function(x)gsub('\\s+', '',x)))
+
+# get full name
+D4 <- unite(D4, "Name", c("First.Name", "Last.Name"), sep = " ")
+
+# create a person-class matrix
+D5 <- gather(D4, "Class", "Class_name", -1, na.rm = TRUE)
+D5$count <- 1
+D6<- pivot_wider(D5, id_cols = "Name", names_from = "Class_name", values_from = "count")
+D6[is.na(D6)] <- 0
+M1 <- as.matrix(D6[,-1])
+rownames(M1) <- D6$Name
+
+# create a person-person matrix
+M2 <- M1 %*% t(M1)
+
+# plot the matrix
+library(igraph)
+g3 <- graph_from_adjacency_matrix(M2, mode = "undirected", weighted = NULL, diag = FALSE)
+plot(g3, layout = layout.fruchterman.reingold)
+
+# generate betweeness centrality and dregree
+betweenness(g3, directed = FALSE, weights = NULL, normalized = FALSE)
+centr_degree(g3)
+
+# The most central person in the network is XUDIAN ZHANG.
```

Comment:
Try to make a substantive meaning from this. As in, what does it mean to be the most central person in this network?